src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.cs

```diff
@@ -86,7 +96,43 @@ public sealed partial class AnalyzerConfig
         /// <summary>
         /// Gets whether this editorconfig is a global editorconfig.
         /// </summary>
-        internal bool IsGlobal => GlobalSection.Properties.ContainsKey(GlobalKey);
+        internal bool IsGlobal => IsGlobalFileName(this.PathToFile) || GlobalSection.Properties.ContainsKey(GlobalKey);
+
+        /// <summary>
+        /// Get the global level of this config, used to resolve conflicting keys
+        /// </summary>
+        /// <remarks>
+        /// A user can explicitly set the global level via the <see cref="GlobalLevelKey"/>.
+        /// When no global level is explicitly set, we use a heuristic:
+        ///  <list type="bullet">
+        ///     <item><description>
+        ///     Any file matching the <see cref="UserGlobalConfigName"/> is determined to be a user supplied global config and gets a level of 100
+        ///     </description></item>
+        ///     <item><description>
+        ///     Any other file gets a default level of 0
+        ///     </description></item>
+        ///  </list>
+        ///  
+        /// This value is unsued when <see cref="IsGlobal"/> is <c>false</c>;
```

Comment:
>This value is unsued when <see cref="IsGlobal"/> is <c>false</c> [](start = 12, length = 64)

Is this correct that the value is unused when `IsGlobal` is `false`? It looks like `MergeSection()` uses this value whenever there is a conflict.

Reply:
Yeah, `MergeSection()` is part of the global config builder, so we'll only get there when `IsGlobal` is true